### PR TITLE
use ArrayDeque, less ram for PacketWrapper.readableObjects

### PR DESCRIPTION
**Pull request by creeper123123321**

### CODE REVIEW

**Comment by tomcraft** [comment]
You could use a reverse Iterator to prevent the Collections.reverse operation.

https://stackoverflow.com/questions/18760832/how-can-i-iterate-a-arraydeque-in-reverse

**Comment by tomcraft** [comment]
Oh no I'm tired

**Comment by tomcraft** [comment]
Oh no wrong field. But you can just iterate with index based loop instead of reversing the whole List

File: common/src/main/java/us/myles/ViaVersion/api/PacketWrapper.java [hunk 1]
```diff
@@ -488,7 +490,10 @@ public UserConnection user() {
      */
     public void resetReader() {
         // Move all packet values to the readable for next packet.
-        this.readableObjects.addAll(0, packetValues);
+        Collections.reverse(packetValues);
+        for (Pair<Type, Object> obj : this.packetValues) {
+            this.readableObjects.addFirst(obj);
+        }
         this.packetValues.clear();
```
Comment (tomcraft) [inline common/src/main/java/us/myles/ViaVersion/api/PacketWrapper.java:497]: for (int i = packetValues.size() - 1; i >= 0; i--) {
            this.readableObjects.addFirst(this.packetValues.get(i));
        }
        this.packetValues.clear();

**Review by KennyTV — APPROVED** [review]

**Pull request merged**